napari/layers/_vectors_layer/model.py

```diff
@@ -0,0 +1,457 @@
+        width of the line in pixels
+            widths greater than 1px only guaranteed to work with "agg" method
+        :return: int or float line width
+        """
+        return self._width
+
+    @width.setter
+    def width(self, width: Union[int, float]) -> None:
+        self._width = width
+        self._refresh()
+
+    @property
+    def length(self) -> Union[int, float]:
+        return self._length
+
+    @length.setter
+    def length(self, length: Union[int, float]):
+        """
+        Change the length of all lines
+        :param length: length multiplicative factor
+        :return: None
+        """
+        self._length = length
+
+        # emit signal back to qt_properties for length
+        self.events.emit_len()
+
+        self._refresh()
+
+    def _default_length(self):
+        """
+        Default method for calculating vector lengths
+        Implemented ONLY for image-like vector data
+        :param event_len: new length
+        :return:
+        """
+
+        if self._data_type == 'coords':
+            return None
+        elif self._data_type == 'image':
+            self._vectors = self._convert_to_vector_type(self._current_data)
+
+    @property
+    def color(self) -> str:
+        """Color, ColorArray: color of the body of the marker
+        """
+        return self._color
+
+    @color.setter
+    def color(self, color: str) -> None:
+        self._color = color
+        self._refresh()
+
+    # =========================== Napari Layer ABC methods ===================
+    @property
+    def data(self) -> np.ndarray:
+        """
+
+        :return: coordinates of line vertices
+        """
+        return self.vectors
+
+    @data.setter
+    def data(self, data: np.ndarray) -> None:
+        """
+        :param data:
+        :return:
+        """
+
+        self.vectors = data
+
+    def _get_shape(self):
+        if len(self.vectors) == 0:
+            return np.ones(self.vectors.shape, dtype=int)
+        else:
+            return np.max(self.vectors, axis=0) + 1
+
+    def _refresh(self):
+        """Fully refresh the underlying visual.
+        """
+        self._need_display_update = True
+        self._update()
+
+    def _update(self):
+        """Update the underlying visual.
+        """
+        if self._need_display_update:
+            self._need_display_update = False
+
+            self._set_view_slice(self.viewer.dims.indices)
+
+        if self._need_visual_update:
+            self._need_visual_update = False
+            self._node.update()
+
+    def _set_view_slice(self, indices):
+        """Sets the view given the indices to slice with.
+
+        Parameters
+        ----------
+        indices : sequence of int or slice
+            Indices to slice with.
+        """
+        
+        in_slice_vectors = self.vectors
+
+        # Display vectors if there are any in this slice
+        if len(in_slice_vectors) > 0:
+            data = np.array(in_slice_vectors) + 0.5
```

Comment:
please explain in a comment as to why you need to add `0.5` to the array

Reply:
this is something that happens in the markers layer too to make the beginning of the line appear at the center of the pixel. we should probably explain there too